lib/linguist/heuristics.yml

```diff
@@ -48,7 +48,13 @@ disambiguations:
 - extensions: ['.as']
   rules:
   - language: ActionScript
-    pattern: '^\s*(package\s+[a-z0-9_\.]+|import\s+[a-zA-Z0-9_\.]+;|class\s+[A-Za-z0-9_]+\s+extends\s+[A-Za-z0-9_]+)'
+    pattern: 'package\s+[\w\.]*\s*[{\n]{1}'
```

Comment:
Are these valid only at the beginning of a string? If that's the case, they should be changed to preserve the previous behaviour, which is more restrictive. For example:
`^\s*package\s+[\w\.]*\s*[{\n]{1}`. This will prevent matching in the middle of statements, within quoted strings, comments, etc.

Reply:
I added anchoring to all the expressions now, so that should avoid matching comments